SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/ReferenceCountedLockingPolicy/ReferenceCountedLockingPolicyTest.cpp

```diff
@@ -93,27 +128,163 @@ bool FWaitForActor::Update()
 	return (IsValid(Actor) && Actor->IsActorInitialized() && Actor->HasActorBegunPlay());
 }
 
-DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FTestIsLocked, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, Handle, bool, bExpected);
+DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FAcquireLock, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, ActorHandle, FString, DebugString);
+bool FAcquireLock::Update()
+{
+	AActor* Actor = Data->TestActors[ActorHandle];
+	const ActorLockToken Token = Data->LockingPolicy->AcquireLock(Actor, DebugString);
+
+	// If the token returned is valid, it MUST be unique
+	if (Token != SpatialConstants::INVALID_ACTOR_LOCK_TOKEN)
+	{
+		for (const TPair<AActor*, TArray<LockingTokenAndDebugString>>& ActorLockingTokenAndDebugStrings : Data->TestActorToLockingTokenAndDebugStrings)
+		{
+			const TArray<LockingTokenAndDebugString>& LockingTokensAndDebugStrings = ActorLockingTokenAndDebugStrings.Value;
+			bool TokenAlreadyExists = LockingTokensAndDebugStrings.ContainsByPredicate([Token](const LockingTokenAndDebugString& Data)
+			{
+				return Token == Data.Key;
+			});
+			if (TokenAlreadyExists)
+			{
+				Test->AddError(FString::Printf(TEXT("AcquireLock returned a valid ActorLockToken that had already been assigned. Token: %d"), Token));
+			}
+		}
+	}
+
+	TArray<LockingTokenAndDebugString>& ActorLockTokens = Data->TestActorToLockingTokenAndDebugStrings.FindOrAdd(Actor);
+	ActorLockTokens.Emplace(TPairInitializer<ActorLockToken, FString>(Token, DebugString));
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FOUR_PARAMETER(FReleaseLock, FAutomationTestBase*, Test, TSharedPtr<TestData>, Data, FName, ActorHandle, FString, LockDebugString);
+bool FReleaseLock::Update()
+{
+	const AActor* Actor = Data->TestActors[ActorHandle];
```

Comment:
Is it possible that the actor isn't in the map? If so we should test for it.

Reply:
Unreal Map [] is similar to unordere_map::at

Reply:
Do you want to do a proper `Test->TestTrue(...)` for this? My thinking is that there'd be this and other things which would produce errors/exceptions and thus fail the test but aren't a proper part of the tested feature functionality (hence not using the Test syntax)